common/static/js/src/string_utils.js

```diff
@@ -45,4 +47,278 @@
         return _.template(text, {interpolate: /\{(.+?)\}/g})(values);
     };
     this.interpolate_text = interpolate_text;
+
+    edx.StringUtils = edx.StringUtils || (function() {
```

Comment:
Good point. I think the right solution is to change HtmlUtils and StirngUtils (in the pattern library) so they install on the window object if RequireJS is not present (using the same syntax that many libraries due, and that the optimizer can handle). Then we list them in base_vendor_js (which means they load BEFORE RequireJS, synchronously) and install them into RequireJS the same way we do with underscore, JQuery, etc.